packages/transport/src/modules/messages/MessageController.ts

```diff
@@ -287,16 +287,26 @@ export class MessageController extends TransportController {
     @log()
     public async sendMessage(parameters: ISendMessageParameters): Promise<Message> {
         const parsedParams = SendMessageParameters.from(parameters);
+
         if (!parsedParams.attachments) parsedParams.attachments = [];
 
         const secret = await CoreCrypto.generateSecretKey();
         const serializedSecret = secret.serialize(false);
         const addressArray: ICoreAddress[] = [];
         const envelopeRecipients: MessageEnvelopeRecipient[] = [];
+        const deletedPeers: string[] = [];
```

Comment:
all this new logic should be extracted to a new **private** method. sendMessage gets way to cluttered here.